### ISSUE TITLE
Hide files (folders) and add them to the ignore list at the same time

### ISSUE THREAD

**Issue by wizardwen** [JonasDoesThings/obsidian-hide-folders#10]
After a file/folder has been hidden, it remains visible in places such as Quick Switch or search despite its hidden status.
 It is hoped that a feature allowing folders to be hidden and added to the system's ignore list can be implemented. 
Currently, folders can be manually added to this ignore list, but if visibility is desired in searches, the ignored folder must be removed from the settings, which is cumbersome. Enabling the addition or removal of folders from the ignore list while toggling visibility would streamline the process, requiring only a single click of the sidebar button or execution of the command. 
Thank you!

**Comment by JonasDoesThings** [comment]
Hmm, seems like I can rather easily modify Obsidian's excluded items list. However I will probably make this into a opt-in feature which users can enable in my plugins settings.

Still looking into it

**Comment by JonasDoesThings** [comment]
Ok FINALLY found some time and motivation to finish this. Released in v1.3.0.

Had to add some extra logic, so when changing other settings the ignore list also gets updated accordingly to minimize cases where leftover "corpses" could end up on the ignore list.

For example when removing a folder from the "hidden folders" settings we need to remove it from ignore list too, when toggling case (in)sensitivity, we need to rewrite the ignore list entries, etc.

Also I finally found out that my initial approach changing obsidian's internal config didn't reload and persist it after changing the ignoreList, this works too now.
